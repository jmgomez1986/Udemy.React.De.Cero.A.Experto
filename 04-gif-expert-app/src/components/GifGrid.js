import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

	const [images, setImages] = useState([]);

	useEffect(() => {
		getGifs();
	}, []); // Al mandar la lista  de dependencias vacia, se ejecta solo en la primera renderizacion

	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=5Aur2asApFfGBynHWjHEVbAzIUjLAKK5`
		const response = await fetch(url);
		const {data} = await response.json();

		const gifs = data.map( (img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url
			}
		});

		setImages(gifs);
	}

	return (
		<>

			<h3>{category}</h3>

			<div className="card-grid">
				
				{
					images.map( (img) => 
						<GifGridItem
							key={img.id}
							// img={img}
							{...img}
						/>
					)
				}

			</div>
			
		</>
	)
}
