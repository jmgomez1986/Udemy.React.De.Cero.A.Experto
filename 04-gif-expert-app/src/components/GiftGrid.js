import React from 'react'

export const GiftGrid = ({category}) => {

	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=5Aur2asApFfGBynHWjHEVbAzIUjLAKK5`
		const response = await fetch(url);
		const {data} = await response.json();

		const gifs = data.map( (img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url
			}
		})

		console.log(gifs);
	}

	getGifs();

	return (
		<div>
			<h3>{category}</h3>
		</div>
	)
}
