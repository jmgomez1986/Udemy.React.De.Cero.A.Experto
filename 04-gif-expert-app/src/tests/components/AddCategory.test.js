import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruevas en el componente <AddCategory />', () => {
	const setCategories = () => {}

	test('debe de mostrarse el componente <AddCategoy /> correctamente ', () => {
		const wrapper = shallow(<AddCategory setCategorias={setCategories} />);

		expect(wrapper).toMatchSnapshot();
	});
	
});
