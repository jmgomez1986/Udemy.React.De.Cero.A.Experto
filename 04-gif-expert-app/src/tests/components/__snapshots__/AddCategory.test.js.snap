// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en el componente <AddCategory /> debe de mostrarse el componente <AddCategoy /> correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
